x-pack/test/api_integration/deployment_agnostic/apis/observability/synthetics/create_monitor.ts

```diff
@@ -0,0 +1,303 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import expect from '@kbn/expect';
+import { RoleCredentials } from '@kbn/ftr-common-functional-services';
+import epct from 'expect';
+import moment from 'moment/moment';
+import { v4 as uuidv4 } from 'uuid';
+import { omit, omitBy } from 'lodash';
+import {
+  ConfigKey,
+  MonitorTypeEnum,
+  HTTPFields,
+} from '@kbn/synthetics-plugin/common/runtime_types';
+import { formatKibanaNamespace } from '@kbn/synthetics-plugin/common/formatters';
+import { SYNTHETICS_API_URLS } from '@kbn/synthetics-plugin/common/constants';
+import { DEFAULT_FIELDS } from '@kbn/synthetics-plugin/common/constants/monitor_defaults';
+import {
+  removeMonitorEmptyValues,
+  transformPublicKeys,
+} from '@kbn/synthetics-plugin/server/routes/monitor_cruds/formatters/saved_object_to_monitor';
+import { DeploymentAgnosticFtrProviderContext } from '../../../ftr_provider_context';
+import { getFixtureJson } from './helpers/get_fixture_json';
+import { SyntheticsMonitorTestService } from '../../../services/synthetics_monitor';
+
+// adjust the type of samlAuth
```

Comment:
I realized I left this as a comment for myself because samlAuth is set to `any` here. Didn't get to fixing it.

Reply:
I like to preface comments like this with `TODO` to signify that the PR has new tech debt to resolve before merging.

It's a nice reminder to the author themselves and it gives a much stronger indicator to reviewers that something is unfinished.